lib/rubocop/ast/processed_source.rb

```diff
@@ -159,6 +160,37 @@ def line_indentation(line_number)
           .length
       end
 
+      def node_tokens(node)
+        @node_tokens ||= {}.compare_by_identity
```

Comment:
I think a worthwhile method would be `def tokens_within(source_range)`.

Was this cache actually helpful in your tests, in the end? If so, then we might not even need `node_tokens`, otherwise we could have both.

Reply:
Retested with and without cache - it is useless so removed that and `node_tokens` method; added `tokens_within`